File: app/src/main/java/com/zires/serverhealthchecker/Application.java
package com.zires.serverhealthchecker;

import com.zires.lib.Notification;
import com.zires.serverhealthchecker.server.DatabaseServer;
import com.zires.serverhealthchecker.server.Server;
import com.zires.serverhealthchecker.server.checker.ServerChecker;

/**
 * Created by ClassicZires on 11/20/2019.
 **/

class Application {
    public Application() {
        Server databaseServer = new DatabaseServer();
        Server server = ServerChecker.Companion.overallHealth(databaseServer);
        Notification notification = HealthLevelParser.INSTANCE.toNotification(server.getHealthLevel());
        notification.send(server.getHealthLevel().name());
    }
}

File: app/src/main/java/com/zires/serverhealthchecker/Application.kt
package com.zires.serverhealthchecker

import com.zires.serverhealthchecker.notification.Notification
import com.zires.serverhealthchecker.server.ServerHealth
import com.zires.serverhealthchecker.server.all.DatabaseServerHealth
import com.zires.serverhealthchecker.server.all.FileServerHealth
import com.zires.serverhealthchecker.server.all.WebServerHealth
import java.util.*

/**
 * Created by ClassicZires on 11/20/2019.
 */

internal class Application {
    init {
        val serversHealth: ArrayList<ServerHealth> = arrayListOf()
        serversHealth.add(DatabaseServerHealth())
        serversHealth.add(WebServerHealth())
        serversHealth.add(FileServerHealth())

        for (serverHealth in serversHealth) {
            var notifications: ArrayList<Notification> = serverHealth.check()
            for (notification in notifications) {
                notification.send(serverHealth)
            }
        }
    }
}
